lib/generators/metka/strategies/materialized_view/templates/migration.rb.erb

```diff
@@ -5,41 +5,48 @@ class <%= @migration_class_name %> < ActiveRecord::Migration<%= ActiveRecord::VE
     execute <<-SQL
     CREATE OR REPLACE FUNCTION metka_refresh_<%= view_name %>_materialized_view() RETURNS trigger LANGUAGE plpgsql AS $$
       BEGIN
-        IF TG_OP = 'INSERT' AND NEW.<%= source_column_name %> IS NOT NULL THEN
+        IF TG_OP = 'INSERT' AND
+           (<%= source_columns.map { |column| "NEW.#{column} IS NOT NULL" }.join(' OR ') %>) THEN
           REFRESH MATERIALIZED VIEW CONCURRENTLY <%= view_name %>;
         END IF;
 
-        IF TG_OP = 'UPDATE' AND OLD.<%= source_column_name %> != NEW.<%= source_column_name %> THEN
+        IF TG_OP = 'UPDATE' AND
+           (<%= source_columns.map { |column| "OLD.#{column} IS DISTINCT FROM NEW.#{column}" }.join(' OR ') %>) THEN
           REFRESH MATERIALIZED VIEW CONCURRENTLY <%= view_name %>;
         END IF;
 
-        IF TG_OP = 'DELETE' AND OLD.<%= source_column_name %> IS NOT NULL THEN
+        IF TG_OP = 'DELETE' AND
+           (<%= source_columns.map { |column| "OLD.#{column} IS NOT NULL" }.join(' OR ') %>) THEN
           REFRESH MATERIALIZED VIEW CONCURRENTLY <%= view_name %>;
         END IF;
         RETURN NEW;
       END $$;
 
     DROP MATERIALIZED VIEW IF EXISTS <%= view_name %>;
     CREATE MATERIALIZED VIEW <%= view_name %> AS
-      SELECT UNNEST
-        ( <%= source_column_name %> ) AS <%= source_column_name.singularize %>_name,
-        COUNT ( * ) AS taggings_count
-      FROM
-        <%= source_table_name %>
+      SELECT
+        tag_name,
+        COUNT(*) AS taggings_count
+      FROM (
+        SELECT UNNEST
+          (<%= source_columns.join(' || ') %>) AS tag_name
+        FROM
+          <%= source_table_name %>
+      ) subquery
       GROUP BY
-        <%= source_column_name.singularize %>_name;
+        tag_name;
```

Comment:
But in that case, if UNNEST is put within subquery, the Query Plan starts looking like this (with exactly the same initial setup):

```
HashAggregate  (cost=4100.00..4102.00 rows=200 width=40) (actual time=4.936..4.951 rows=100 loops=1)
  Group Key: unnest((constructs.tags || constructs.materials))
  ->  ProjectSet  (cost=0.00..1100.00 rows=200000 width=32) (actual time=0.020..2.555 rows=11975 loops=1)
        ->  Seq Scan on constructs  (cost=0.00..80.00 rows=2000 width=117) (actual time=0.011..0.465 rows=2000 loops=1)
Planning Time: 0.358 ms
Execution Time: 5.022 ms
```

Which already looks much better